examples/imagenet.py

```diff
@@ -0,0 +1,363 @@
+import os
+import warnings
+
+import hydra
+import torch
+import torch.nn as nn
+import torch.backends.cudnn as cudnn
+import torch.distributed as dist
+import torchvision.models as models
+import torchvision.datasets as datasets
+import torchvision.transforms as transforms
+from torch.optim import Adam
+
+best_acc1 = 0
+
+
+@hydra.main(config_name="imagenetconf")
+def main(cfg):
+    if cfg.seed is not None:
+        random.seed(cfg.seed)
+        torch.manual_seed(cfg.seed)
+        cudnn.deterministic = True
+        warnings.warn(
+            "You have chosen to seed training. "
+            "This will turn on the CUDNN deterministic setting, "
+            "which can slow down your training considerably! "
+            "You may see unexpected behavior when restarting "
+            "from checkpoints."
+        )
+    if cfg.gpu is not None:
+        warnings.warn(
+            "You have chosen a specific GPU. This will completely "
+            "disable data parallelism."
+        )
+    if cfg.dist_url == "env://" and cfg.world_size == -1:
+        cfg.world_size = int(os.environ["WORLD_SIZE"])
+
+    global best_acc1
+
+    if cfg.gpu is not None:
+        print("Use GPU: {} for training".format(cfg.gpu))
```

Comment:
1. Use Python logging instead of print() per the examples in Hydra.
2. Use Python 3 fstrings.

Reply:
Will do! Just curious, may I ask why you prefer fstrings over format?

Reply:
It's more readable.
compare:
```python
print("Use GPU: {} for training".format(cfg.gpu))
```
with
```python
print(f"Use GPU: {cfg.gpu} for training")
```